src/Tribe/Shortcode/Shortcode_Interface.php

```diff
@@ -0,0 +1,114 @@
+<?php
+/**
+ * The interface all shortcodes should implement.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Shortcode
+ */
+namespace Tribe\Shortcode;
+
+/**
+ * Interface Shortcode_Interface
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Shortcode
+ */
+interface Shortcode_Interface {
+
+	/**
+	 * Returns the shortcode slug.
+	 *
+	 * The slug should be the one that will allow the shortcode to be built by the shortcode class by slug.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string The shortcode slug.
+	 */
+	public function get_registration_slug();
+
+	/**
+	 * Configures the base variables for an instance of shortcode.
+	 *
+	 * @since  TBD
+	 *
+	 * @param array  $arguments Set of arguments passed to the Shortcode at hand.
+	 * @param string $content   Contents passed to the shortcode, inside of the open and close brackets.
+	 *
+	 * @return void
```

Comment:
Remove this, same "we cannot really add a meaningful doc block here" question as above.